### DISCUSSION TITLE
Allow passing lambda for `list_sort` for sorting list of structs

### DISCUSSION THREAD

**Discussion by Swoorup** [duckdb/duckdb#5623] [Ideas]
Why do you want this feature?

To be able to sort a list of `STRUCT` data type, using a particular field in the struct.

Example: 

```sql
select list_sort(l, x -> x.price) from event;
```

**Comment by taniabogatsch** [comment]
```sql
select list_apply(list_sort(list_apply(col, x -> (x.price, x)), 'DESC'), y -> y.x) from my_table;
```
This could be a workaround for now. But I will keep this feature idea in mind for when I pick up work on the lists and lambdas again.
😄

**Reply by lnkuiper** [reply]
Just documenting this here:
```sql
select [y -> y.x for y in list_sort([x -> (x.price, x) for x in my_list], 'DESC')] from my_table;
Error: Binder Error: No function matches the given name and argument types 'json_extract(STRUCT(id BIGINT, price DOUBLE), STRUCT(v1 DOUBLE, x STRUCT(id BIGINT, price DOUBLE)))'. You might need to add explicit type casts.
    Candidate functions:
    json_extract(JSON, VARCHAR) -> JSON
    json_extract(JSON, VARCHAR[]) -> JSON[]
```

This should work

**Reply by lnkuiper** [reply]
Oops, this is just a syntax error on my part. It works:
```sql
select [y.x for y in list_sort([(x.price, x) for x in my_list], 'DESC')] from my_table;
```

**Comment by l1t1** [comment]
list_sort([(x.price, x)  for x in my_list], 'DESC')]
means
order by (x.price, x)  desc?
how to order by x. price desc ,x asc?

**Comment by taniabogatsch** [comment]
At this point, I'd maybe just go for a staging table to utilize the actual `ORDER BY` clause. Ofc a hacky solution is possible, you could temporarily invert `x`, but yea, nested types and the functions that we have to support them are not as extensive as the relational operators for tables.
